runtime/runtime/src/lib.rs

```diff
@@ -289,6 +290,28 @@ impl Runtime {
         debug!(target: "runtime", "{}", log_str);
     }
 
+    /// Validates all transactions in parallel and returns an iterator of
+    /// transactions paired with their validation results.
+    ///
+    /// Returns an `Iterator` of `(&SignedTransaction, Result<TransactionCost, InvalidTxError>)`
+    fn parallel_validate_transactions<'a>(
+        config: &'a RuntimeConfig,
+        gas_price: Balance,
+        transactions: &'a [SignedTransaction],
+        current_protocol_version: ProtocolVersion,
+    ) -> impl Iterator<Item = (&'a SignedTransaction, Result<TransactionCost, InvalidTxError>)>
+    {
+        let results: Vec<_> = transactions
+            .par_iter()
+            .map(move |tx| {
+                let cost_result =
+                    validate_transaction(config, gas_price, tx, true, current_protocol_version);
+                (tx, cost_result)
+            })
+            .collect();
+        results.into_iter()
```

Comment:
```suggestion
        transactions
            .par_iter()
            .map(move |tx| {
                let cost_result =
                    validate_transaction(config, gas_price, tx, true, current_protocol_version);
                (tx, cost_result)
            })
```

Reply:
If `par_iter` does not become an `impl Iterator`, feel free to return `impl rayon::ParallelIterator` or whatever, but there really is no reason to go through a vector here.

Reply:
Ah, never mind me, looks like you can't actually get out of parallel iteration without collecting into a structure currently. My bad.